oio/rebuilder/meta_rebuilder.py

```diff
@@ -67,32 +67,38 @@ def __init__(self, conf, logger, type_, max_attempts=5, **kwargs):
         super(MetaRebuilderWorker, self).__init__(conf, logger, **kwargs)
         self.type = type_
         self.max_attempts = max_attempts
-        self.proxy_client = ProxyClient(
-            self.conf, request_prefix='/admin', logger=self.logger)
+        self.admin_client = AdminClient(self.conf, logger=self.logger)
 
     def _rebuild_one(self, cid, **kwargs):
-        attempts = 0
-        while True:
-            attempts += 1
-            params = {'cid': cid, 'type': self.type}
+        missing_base = False
+        for attempts in range(self.max_attempts):
             try:
-                properties = {'properties': {'sys.last_rebuild':
-                                             str(int(time.time()))}}
-                # Setting a property will trigger a database replication
-                self.proxy_client._request('POST', '/set_properties',
-                                           params=params, json=properties)
+                if not missing_base:
+                    # Check if the bases exist
+                    info = self.admin_client.has_base(self.type, cid=cid)
+                    for meta in info.values():
+                        if meta['status']['status'] != 200:
+                            missing_base = True
+                            break
+
+                if missing_base:
+                    # Setting a property will trigger a database replication
```

Comment:
Would it be better to also ask for an `/election/exit` before adding the property?